src/filesystem.cc

```diff
@@ -132,8 +132,8 @@ class PosixReadableFile : public ReadableFile {
 
   bool ReadLineStdin(ps_string *line) {
     if (stdin_) {
-      std::string tmp;
-      auto worked = static_cast<bool>(std::getline(std::cin, tmp, delim_));
+      std::shared_ptr<std::string> tmp = std::make_shared<std::string>();
+      auto worked = static_cast<bool>(std::getline(std::cin, *tmp, delim_));
```

Comment:
```suggestion
const bool worked = std::getline(std::cin, *tmp, delim_);
```

`std::getline` has a conversion operator to `bool`, which means we can avoid the cast. The semantics are the same.